Upgrade to `cobra` 4.8 to support space-ros upgrade to Jazzy/Noble (closes #20)

asimonov opened the pull request.

asimonov commented:
ok, Cobra has fixed build process in 4.8 and I verified this change works space-ros/jazzy

@ivanperez-keera can you review and merge?

ivanperez-keera commented:
@asimonov and I just reviewed this together. We'd like to make the following changes before the PR is merged:
- Remove the commented code.
- Move the comment associated to the commented code to the commit message contents.
- Make sure the VSCode plugin can parse the new output, since the format of Cobra-produced (SARIF) JSON files appears to have changed.

asimonov committed:
upgrade cobra_vendor to cobra 4.8 (issue #20)

Upgrade Cobra to version 4.8.

This is part of space-ros upgrade to Jazzy/Noble (Ubuntu24.04).

Tested with both Jazzy and Humble versions of space-ros:
* build works
* cobra tests run
* output of cobra is slightly different for MISRA ruleset due to message description copyright issues

Cobra 4.8 make install is now putting files directly to /usr/local/bin, so we have to use sudo.
Also removed the install part of cmake as bin_linux directory is not generated due to direct install.

asimonov commented:
> @asimonov and I just reviewed this together. We'd like to make the following changes before the PR is merged:
> 
> * Remove the commented code.
> * Move the comment associated to the commented code to the commit message contents.
> * Make sure the VSCode plugin can parse the new output, since the format of Cobra-produced (SARIF) JSON files appears to have changed.

@ivanperez-keera, I have re-done the commit with the agreed changes.

I dont have capacity to check the output with VScode plugin. But at least it builds/runs on both Humble and Jazzy

lets decide if it should be on new release branch rather than `master` and I can amend the right branch
